apps/whale-api/src/module.api/governance.service.ts

```diff
@@ -99,8 +102,10 @@ export class GovernanceService {
       }))
 
       return ApiPagedResponse.of(votes, size, () => {
-        if (next === undefined) {
+        if (next === undefined && votes.length > 0) {
           return (votes.length - 1).toString()
+        } else if (next === undefined) {
+          return undefined
         }
```

Comment:
```suggestion
        if (next === undefined) {
          if (votes.length > 0) {
            return (votes.length - 1).toString()
          }
          return undefined
        }
```

Prehaps?

Reply:
wanted to not use nested loops lol, but sure will update

Reply:
Logically this reduces the `if + &`  from 3 to 2, and it's easier to follow. 

Initially, when I read it, I saw `next === undefined`; my brain will automatically think all conditions where `next is undefined` will be handled in this code branch only. But when I saw another `next === undefined` scenario in the next `if` statement, it can hard to follow since the conditions flow like "switch-cases" instead of "logical branches".

"switch-cases" you have the revisit all conditions and reevaluate each case, while for logical branches, based on the `{}` scopes, you can easily map out within this branch/scope all conditions here are when `next is undefined`.

> 🥹